### ISSUE TITLE
can you please explain (x/8,y/8,1).

### ISSUE THREAD

**Issue by norahik** [Neerajj9/CSRNet-keras#5]
can you pleas eexplain why the the desnity map of the output is of size (x/8,y/8,1).

**Comment by aditya-malte** [comment]
The size is reduced due to the 3 pooling layers.
Each one reducing the matrix size by a factor of 2.
Thanks

**Comment by norahik** [comment]
thank you for your response on such a novice question.
if you allow, ( or i can delete this comment) i have a general question regarding creating a density map out of ground-truth:
why do we have to measure the k-nearest neighbor? it the ground-truth represent head coordinates, can't we just create an array of zeros the same size of the image and label each head coordinate with 1?

**Comment by aditya-malte** [comment]
Good question! 
The math is actually quite interesting.
Directly training on the discrete 1s wouldn't lead to convergence of the CNN.
So we use a gaussian convolution to make it continuous. The sigma of the convolution is chosen such that it is approximately representative of the size of that head(deduced by taking the average of the k nearest neighbors and multiplying it by 0.3). You'll get a similar explanation from the Zang et. al. paper of crowd counting.
Thanks

**Comment by norahik** [comment]
thank you.

to get my understanding right, if i want to segment the original image,, therefor i have to generate the full "hot encoded matrix" directly from the full image then segment the image and the "hot encoded matrix" in the same manner ,,, then pass each segment to the "gaussian_filter_density" method as if the image segment and the hot encoded matrix segment are new full image and matrix of their own.

**Comment by norahik** [comment]
sorry maybe you got may question wrong. 
as you said your network takes variable image sizes as input. 
for my case i want the input size to be fixed, there for i have to segment the dataset images into pre-set dimensions say 300x300. there for the ground truth also need to be segmented so that each image has its representative dot-map. the resulting images and dot-maps will be as a new dataset for data augmentation reasons.

**Comment by aditya-malte** [comment]
Hello,
Yes, if you intend to augment/crop the images then you also need to crop the ground truth too correspondingly.
Thank you
